### PR TITLE
USHIFT-362: consolidate all build output directories

### PR DESCRIPTION
**Pull request by dhellmann**
Move all of the build output under an `_output` directory at the root
of the repository to make it easier to clean it all up at one time and
to make the .gitignore file easier to manage.

/assign @fzdarsky

### CODE REVIEW

**Review by fzdarsky — APPROVED** [review]
Updated the `make clean` target, too. Now LGTM.

File: .gitignore [hunk 1]
```diff
@@ -1,14 +1,7 @@
 .idea/
 .vagrant/
 .vscode/
-/microshift
-/paack-result/
 Vagrantfile
```
Comment (ggiguash) [inline .gitignore:4]: Don't we need to keep /microshift as the exe still goes to the project root dir, right?

File: docs/devenv_rhel8.md [hunk 2]
```diff
@@ -85,10 +85,10 @@ make srpm
 The artifacts of the build are located in the `packaging` directory.
```
Comment (ggiguash) [inline docs/devenv_rhel8.md:85]: Need to fix the sentence above - the build artifacts are no longer located in `packaging` dir

File: packaging/rpm/make-rpm.sh [hunk 3]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env bash
 set -e -o pipefail
+set -x
```
Comment (ggiguash) [inline packaging/rpm/make-rpm.sh:3]: Should this be removed?

File: packaging/rpm/make-rpm.sh [hunk 4]
```diff
@@ -17,15 +18,15 @@ GIT_SHA=$(git rev-parse HEAD)
...
 TARBALL_FILE="microshift-${GIT_SHORTHASH}.tar.gz"
-RPMBUILD_DIR="${SCRIPT_DIR}/_rpmbuild/"
+RPMBUILD_DIR="$(git rev-parse --show-toplevel)/_output/rpmbuild/"
 
 SOURCE_GIT_TAG="$(git describe --tags | sed s/nightly-/nightly-$(git show -s --format=%ct)_/g )"
 
 
 create_local_tarball() {
-  tar -czf "${RPMBUILD_DIR}/SOURCES/${TARBALL_FILE}" \
+  tar -czvf "${RPMBUILD_DIR}/SOURCES/${TARBALL_FILE}" \
```
Comment (ggiguash) [inline packaging/rpm/make-rpm.sh:27]: Should we remove the verbose option?

**Comment by ggiguash** [comment]
@dhellmann  Tests are failing because they expect RPMs in a different location. It looks like we would need to fix the tests as well.

**Comment by fzdarsky** [comment]
@copejon @mangelajo @sosiouxme for awareness, as this may have side-effects on CI and release.

**Comment by fzdarsky** [comment]
> @dhellmann Tests are failing because they expect RPMs in a different location. It looks like we would need to fix the tests as well.

I've opened a PR which we'll need to merge first: https://github.com/openshift/release/pull/32076

File: docs/devenv_rhel8.md [hunk 5]
```diff
@@ -85,10 +85,10 @@ make srpm
...
 ```bash
 $ find ~/microshift/packaging -name \*.rpm
```
Comment (ggiguash) [inline docs/devenv_rhel8.md:87]: Suggesting:
```
$ find ~/microshift/_output/rpmbuild/ -name \*.rpm
```

Comment (ggiguash) [inline docs/devenv_rhel8.md:87]:
```suggestion
$ find ~/microshift/_output/rpmbuild -name \*.rpm
```

File: docs/devenv_rhel8.md (hunk 2 above)
Comment (ggiguash) [inline docs/devenv_rhel8.md:85]:
```suggestion
The artifacts of the build are located in the `_output/rpmbuild` directory.
```

**Comment by ggiguash** [comment]
I'm approving this and putting on hold. @fzdarsky please unhold when CI is fixed
/lgtm
/hold

**Pull request merged**